src/app/file/bmp_format.cpp

```diff
@@ -545,27 +867,33 @@ static void read_rle4_compressed_image(FILE *f, Image *image, const BITMAPINFOHE
 }
 
 static int read_bitfields_image(FILE *f, Image *image, BITMAPINFOHEADER *infoheader,
-                                unsigned long rmask, unsigned long gmask, unsigned long bmask)
+                                uint32_t rmask, uint32_t gmask, uint32_t bmask,
+                                uint32_t amask, bool& withAlpha)
 {
-#define CALC_SHIFT(c)                           \
-  mask = ~c##mask;                              \
-  c##shift = 0;                                 \
-  while (mask & 1) {                            \
-    ++c##shift;                                 \
-    mask >>= 1;                                 \
-  }                                             \
-  if ((c##mask >> c##shift) == 0x1f)            \
-    c##scale = scale_5bits_to_8bits;            \
-  else if ((c##mask >> c##shift) == 0x3f)       \
-    c##scale = scale_6bits_to_8bits;            \
-  else                                          \
-    c##scale = NULL;
-
-  unsigned long buffer, mask, rshift, gshift, bshift;
-  int i, j, k, line, height, dir, r, g, b;
-  int (*rscale)(int);
-  int (*gscale)(int);
-  int (*bscale)(int);
+#define CALC_SHIFT(c)      \
```

Comment:
As a final comment we can try to replace/refactor these kind of macros if possible (they are from pretty old code).